spatialos-sdk/src/worker/component.rs

```diff
@@ -409,9 +418,20 @@ unsafe extern "C" fn vtable_component_update_deserialize<C: Component>(
     handle_out: *mut *mut Worker_ComponentUpdateHandle,
 ) -> u8 {
     let schema_update = SchemaComponentUpdate::from_raw(update);
-    let deserialized_update = schema_update.deserialize::<C>();
-    *handle_out = handle_allocate(deserialized_update);
-    1
+    match schema_update.deserialize::<C>() {
+        Ok(deserialized_update) => {
+            *handle_out = handle_allocate(deserialized_update);
+            1
+        }
+
+        // TODO: How should we handle errors occurring during vtable serialization? We could
+        // probably store the whole `Result` in the user handle and pass it to the user when
+        // they try to retrieve the result.
```

Comment:
Agree with the suggestion here. 😄

Reply:
Yeah, the more I think about it the more that seems like the right solution. I want to save that for a separate PR because it's going to require that I make some careful changes to how we handle vtable serialization.